src/moscot/base/problems/_mixins.py

```diff
@@ -272,6 +272,47 @@ def _cell_transition_online(
             forward=forward,
         )
 
+    def _celltype_mapping(
+        self: AnalysisMixinProtocol[K, B],
+        mapping_mode: Literal["sum", "max"],
+        key: Optional[str],
+        source: K,
+        target: K,
+        source_groups: Str_Dict_t,
+        target_groups: Str_Dict_t,
+        forward: bool = False,  # return value will be row-stochastic if forward=True, else column-stochastic
+        aggregation_mode: Literal["annotation", "cell"] = "annotation",
+        other_key: Optional[str] = None,
+        other_adata: Optional[str] = None,
+        batch_size: Optional[int] = None,
+        normalize: bool = True,
+        scale_by_marginals: bool = True,
+    ):
+        if mapping_mode == "sum":
+            return self._cell_transition_online(
+                key=key,
+                source=source,
+                target=target,
+                source_groups=source_groups,
+                target_groups=target_groups,
+                forward=forward,
+                aggregation_mode=aggregation_mode,
+                other_key=other_key,
+                other_adata=other_adata,
+                batch_size=batch_size,
+                normalize=normalize,
+            )
+        if mapping_mode == "max":
+            source_annotation_key, source_annotations, source_annotations_ordered = _validate_args_cell_transition(
+                self.adata, source_groups
+            )
+            target_annotation_key, target_annotations, target_annotations_ordered = _validate_args_cell_transition(
+                self.adata if other_adata is None else other_adata, target_groups
+            )
+            dummy = pd.get_dummies(source_annotations)
+            out = self.pull(dummy, scale_by_marginals=scale_by_marginals)
+            return pd.Categorical([dummy.columns[i] for i in np.array(out.argmax(1))])
+
```

Comment:
missing: 'raise NotImplementedError', otherwise the function won't do anything if none of the 'if' conditions are fulfilled.